### PR TITLE
[#25541] Stop use of self hosted runners for some workflows.

### PR DESCRIPTION
**Pull request by lostluck**
Stop use of self hosted runners for some workflows. This is a stopgap to get the repo automation healthy again.

See #25541 for context.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @damccorm

File: .github/workflows/git_tag_released_version.yml
Comment (damccorm) [inline .github/workflows/git_tag_released_version.yml:35]: Its not totally clear if these release tasks will work on hosted runners (or if they'll work at all), but its probably fine to include them. I may try these out during my release

**Review by damccorm — APPROVED** [review]

**Comment by damccorm** [comment]
GitHub won't pick up workflow changes until they're merged, so LabelPrs will still not work on this PR - just calling that out so that you don't block on it

File: .github/workflows/git_tag_released_version.yml
Comment (lostluck) [reply .github/workflows/git_tag_released_version.yml:35]: I thought this was the bot that tagged finished issues/PRs with the release that includes them, and since a release just went out...

**Comment by lostluck** [comment]
And now we wait.

File: .github/workflows/git_tag_released_version.yml
Comment (damccorm) [reply .github/workflows/git_tag_released_version.yml:35]: No, that one is different - this is only run manually and can do one of the release steps (tagging a released version in GitHub) - https://github.com/apache/beam/blob/fac67922cad99979aa6bba5339616fd253e932a7/.github/workflows/git_tag_released_version.yml#L24 (workflow_dispatch means manual trigger only). You're thinking of https://github.com/apache/beam/blob/20c8d3a2ba11f3965022e2ba5e12a241a4ec3442/.github/workflows/assign_milestone.yml#L20

Comment (damccorm) [reply .github/workflows/git_tag_released_version.yml:35]: Fortunately, that should be functioning fine since its on a hosted runner

**Pull request merged**
